images/rclone/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM golang:1.10-alpine3.7 AS build
+
+COPY rclone /go/src/github.com/ncw/rclone/
+COPY patches /go/src/github.com/ncw/rclone/patches
+WORKDIR /go/src/github.com/ncw/rclone/
+
+RUN for file in patches/*.diff ; do \
+        patch -p 1 < "$file" || exit 1 ; \
+    done && \
+    CGO_ENABLED=0 go build -a -installsuffix cgo && \
+    mv rclone /bin/rclone
+
+FROM golang:alpine AS dev
```

Comment:
What's the use of this intermediate containers?

Reply:
Build provides an environment to build the binary.

Dev's purpose is to have a lightweight version of build for testing and debugging purposes. It's not intermediate. It's a different target.